src/components/DnsSettingsForm.js

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) [2020] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useState } from 'react';
+import cockpit from 'cockpit';
+import {
+    FormGroup,
+    TextInput
+} from '@patternfly/react-core';
+import { useNetworkDispatch, updateDnsSettings } from '../context/network';
+import ModalForm from './ModalForm';
+import IPInput from './IPInput';
+
+const _ = cockpit.gettext;
+
+const DnsSettingsForm = ({ isOpen, onClose, dns }) => {
+    const dispatch = useNetworkDispatch();
+    const originalNameServers = (dns?.nameServers || "").split(" ");
```

Comment:
I would rather prefer receiving an array here (not having to split it at this point).